source/tutorials/first-steps/nix-language.md

```diff
@@ -1,28 +1,53 @@
 (reading-nix-language)=
 
-# Nix language basics
-
-The Nix language is used to declare packages and configurations to be built by [Nix][nix-manual].
+# Reading the Nix language without fear
 
+The Nix language is designed for conveniently creating and composing *derivations* – precise descriptions of how contents of existing files are used to derive new files.
 It is a domain-specific, purely functional, lazily evaluated, dynamically typed programming language.
 
-Notable uses of the Nix language are:
+:::{Notable uses of the Nix language}
 
-- [Nixpkgs][nixpkgs]
+- {term}`Nixpkgs`
 
   The largest, most up-to-date software distribution in the world, and written in the Nix language.
 
-- [NixOS][nixos-manual]
+- {term}`NixOS`
 
   A Linux distribution that can be configured fully declaratively and is based on Nix and Nixpkgs.
 
   Its underlying modular configuration system is written in the Nix language, and uses packages from Nixpkgs.
   The operating system environment and services it provides are configured with the Nix language.
 
-[nix-manual]: https://nixos.org/manual/nix/stable
-[nixpkgs-manual]: https://nixos.org/manual/nixpkgs/stable/#preface
-[nixos-manual]: https://nixos.org/manual/nixos/stable/index.html#preface
-[home-manager]: https://github.com/nix-community/home-manager
+:::
+
+You may quickly encounter Nix language expressions that look very complicated.
+As with any programming language, the required amount of Nix language code closely matches the complexity of the problem it is supposed to solve, and reflects how well the problem – and its solution – is understood.
+Building software is a complex undertaking, and Nix both *exposes* and *allows managing* this complexity with the Nix language.
+
+The Nix language has only few basic constructs which can be combined arbitrarily:
+
+- Primitive data types
+
+  such as integers or character strings
+
+- Compound data types
+
+  that is, lists and attribute sets
+
+- Functions and operators
+
+  to produce and transform data
+
+- Name assignment
+
+  to manipulate data as units
```

Comment:
I suppose this refers to scope and perhaps `let` bindings specifically?

```suggestion
- Name assignment

  to reuse data, functions and function arguments
```

I don't quite understand the intent behind these "units", as I can't imagine how naming something would be possible without it being unitary in some given context.

Maybe I'm [Mann-Gelling](https://freethoughtblogs.com/pharyngula/2023/06/19/today-i-learned-about-gell-mann-amnesia/)?

Reply:
I think the original intent behind the "Elements" is rather syntactic, and names are syntactic units, allowing you to refer concisely to semantic units (the things we assign names to).